src/components/RegisterForm/index.tsx

```diff
@@ -0,0 +1,224 @@
+import React, { useRef, useState } from 'react'
+import { Route, Switch, useHistory } from 'react-router-dom'
+import {
+	Errors,
+	Form,
+	FormButton,
+	FormCheckbox,
+	FormInput,
+	FormItem,
+	HiddenError,
+	Label,
+	PasswordMessage,
+	ValidPassword,
+	RegistrationMessage,
+} from './styled'
+
+// library for creating forms
+import { useForm } from 'react-hook-form'
+
+type FormData = {
+	email: string
+	nickname: string
+	password: string
+	passwordConfirm: string
+	check: boolean
+}
+
+const RegisterForm: React.FC = () => {
+	const {
+		register,
+		handleSubmit,
+		errors,
+		watch,
+		setValue,
+	} = useForm<FormData>({
+		mode: 'onChange',
+		reValidateMode: 'onChange',
+	})
+
+	// handle submitting form
+	const changeLocation = useHistory()
+
+	const onSubmit = handleSubmit(
+		() => changeLocation.push('/register/successful')
+
+		// TODO: HANDLE ERRORS
+	)
+
+	// password and PasswordMessage validation
+	const [hidden, setHidden] = useState<boolean>(true)
+
+	const password = useRef('')
+	password.current = watch('password', '')
+
+	const hasLowerCase = /[a-z]/.test(password.current)
+	const hasMinimumLength = password.current.length >= 6
+	const hasUpperCase = /[A-Z]/.test(password.current)
+	const hasNumber = /\d/.test(password.current)
+
+	// nickname prompter
+	const email = useRef('')
+	email.current = watch('email', '')
+
+	const newNickname = email.current.substring(
+		0,
+		email.current.lastIndexOf('@')
+	)
+
+	return (
+		<Switch>
+			<Route exact path="/register">
+				<Form onSubmit={onSubmit}>
```

Comment:
If you add attribute [novalidate](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form#attr-novalidate)  to a form you can better relay on your custom validations. :wink: